lambda/comprehend_pii_sync/app/sync_main.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+import os
+import time
+import boto3
+import textractmanifest as tm
+
+from botocore.exceptions import ClientError
+from typing import List, Tuple
+
+logger = logging.getLogger(__name__)
+version = "0.0.1"
+
+s3 = boto3.client('s3')
+step_functions_client = boto3.client(service_name='stepfunctions')
+comprehend = boto3.client("comprehend")
+
+
+def split_s3_path_to_bucket_and_key(s3_path: str) -> Tuple[str, str]:
+    if len(s3_path) > 7 and s3_path.lower().startswith("s3://"):
+        s3_bucket, s3_key = s3_path.replace("s3://", "").split("/", 1)
+        return (s3_bucket, s3_key)
+    else:
+        raise ValueError(
+            f"s3_path: {s3_path} is no s3_path in the form of s3://bucket/key."
+        )
+
+
+def get_file_from_s3(s3_path: str, range=None) -> bytes:
+    s3_bucket, s3_key = split_s3_path_to_bucket_and_key(s3_path)
+    if range:
+        o = s3.get_object(Bucket=s3_bucket, Key=s3_key, Range=range)
+    else:
+        o = s3.get_object(Bucket=s3_bucket, Key=s3_key)
+    return o.get('Body').read()
+
+
+def send_failure_to_step_function(exception_class,
+                                  error="not_handled_exception",
+                                  cause=None,
+                                  token=None,
+                                  message=None):
+    if not cause:
+        cause = str(exception_class)
+
+    try:
+        step_functions_client.send_task_failure(taskToken=token,
+                                                error=error,
+                                                cause=cause)
+    except step_functions_client.exceptions.InvalidToken:
+        logger.error(f"InvalidToken for message: {message} ")
+    except step_functions_client.exceptions.TaskDoesNotExist:
+        logger.error(f"TaskDoesNotExist for message: {message} ")
+    except step_functions_client.exceptions.TaskTimedOut:
+        logger.error(f"TaskTimedOut for message: {message} ")
+
+
+def send_success_to_step_function(response, token, event, start_time):
+    logger.debug(f"comprehend result: {response}")
+
+    classification_result = "NONE"
+    for c in response['Entities']:
```

Comment:
Input: Textract JSON
Output - PII:
- Enhanced Textract json (list of blocks: words and lines)
- Indicate which words are PII
- Remapping using BeginOffset 
-

Reply:
https://github.com/aws-samples/amazon-textract-idp-cdk-constructs/blob/3f40a57261d75c7a37d9a375472c816327fcad93/lambda/generatecsv/app/main.py#L96